test/dsim/test_shared_array.py

```diff
@@ -17,13 +17,18 @@
 """Unit tests for :mod:`.shared_array`."""
 
 import multiprocessing
-from typing import Generator
+from typing import Generator, Union
 
 import numpy as np
 import pytest
 
 from katgpucbf.dsim.shared_array import SharedArray
 
+# Process isn't an attribute of BaseContext but of each subclass
+_MPContext = Union[
+    multiprocessing.context.ForkContext, multiprocessing.context.ForkServerContext, multiprocessing.context.SpawnContext
+]
+
```

Comment:
Can you not use `|` instead of `Union` in 3.10?

Reply:
For now I'm trying to make it still work in 3.8, until we agree to abandon that.